storage/innobase/dict/dict0dd.cc

```diff
@@ -944,6 +919,40 @@ int dd_table_open_on_dd_obj(dd::cache::Dictionary_client *client,
   return (1);
 }
 
+int dd_table_open_on_dd_obj(dd::cache::Dictionary_client *client,
+                            space_id_t space_id,
+                            const dd::Table &dd_table,
+                            dict_table_t *&table, THD *thd,
+                            const dd::String_type *schema_name,
+                            bool implicit) {
+
+  uint64 table_id = dd_table.se_private_id();
+
+  if (table_id == dd::INVALID_OBJECT_ID) {
+    /* Partitioned table */
+    ut_ad(dd_table_is_partitioned(dd_table));
+    
+    for (auto part : dd_table.leaf_partitions()) {
+      uint64 tid = part->se_private_id();
+      int ret = dd_table_load_part(tid, dd_table, part, table, thd, schema_name, implicit);
+      if(ret != 0) {
+        return ret;
+      }
+      dd_table_close(table, thd, nullptr, false);
+    }
+    return 0;
+  } else {
+    const dd::Partition *dd_part = nullptr;
+    int ret = dd_table_load_part(table_id, dd_table, dd_part, table, thd, schema_name, implicit);
+    if (ret != 0) {
+      return ret;
+    }
+    dd_table_close(table, thd, nullptr, false);
```

Comment:
This looks inconsistent in terms of naming... `dd_table_open_on_dd_obj` should not actually close the table.